packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/MetadataAttributesTable/index.tsx

```diff
@@ -36,6 +42,52 @@ const Th = ({ className, children }: CellProps) => (
   </th>
 );
 
+function ClampedText({ text }: { text: string }) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [hasOverflow, setHasOverflow] = useState(false);
+  const [hasBigword, setHasBigword] = useState(false);
+  const textRef = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    if (textRef.current) {
+      const numLines = textRef.current?.getBoundingClientRect().height / 16;
+      setHasOverflow(numLines >= 4);
+    }
+    setHasBigword(text.split(" ").some((word) => word.length > 20));
```

Comment:
😄 creative solution! Are we considering 20 characters for any specific reason here? Curious to understand the heuristic a bit more. Also, does this really need a separate state for it? It seems to me you could skip this layout effect altogether and have consts for your overflow/long word states

Reply:
I figured that 20 would be a reasonable size for a large word/hash, I confess that I didn't get too attached to this calculation. 

I tried to do it without any effect but without success